hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -656,7 +657,12 @@ private ReferenceCounted<OmSnapshot> getSnapshot(
       // don't allow snapshot indicator without snapshot name
       throw new OMException(INVALID_KEY_NAME);
     }
-
+    // Updating the volumeName & bucketName in case the bucket is a linked bucket. We need to do this before a
+    // permission check, since linked bucket permissions and source bucket permissions could be different.
+    ResolvedBucket resolvedBucket = ozoneManager.resolveBucketLink(Pair.of(volumeName,
```

Comment:
This resolve link involve acl validation, even for internal operation. We should do acl validation for all external operation.
IMO, we should not propogate linked bucket details to other tables, and should be resolved at external operation entry points only.

Reply:
Yup, Nice catch ! I have done it. Can you check this once.